lib/resources/http.rb

```diff
@@ -0,0 +1,51 @@
+# encoding: utf-8
+# copyright: 2017, Criteo
+# author: Guilhem Lettron
+# license: All rights reserved
+
+require 'requests'
+
+module Inspec::Resources
+  class Http < Inspec.resource(1)
+    name 'http'
+    desc 'Use the http InSpec audit resource to test http call.'
+    example "
+      describe http(http://localhost:8080/ping', auth: ['user, 'test'], params: {format: 'html'}) do
+        its('status') { should eq(200) }
+        its('body') { should eq('pong')}
+        its('Content-Type') { should include('text/html') }
+      end
+    "
+
+    # rubocop:disable ParameterLists
+    def initialize(url, method: 'GET', params: nil, auth: nil, headers: {}, data: nil)
+      @method = method
+      @url = url || 'http://localhost/'
+      @params = params
+      @auth = auth
+      @headers = headers
+      @data = data
+      @request = Requests.request(@method, @url,
```

Comment:
as said in other comment, doing stuff in `Net::HTTP` is really painful.
_https_, _headers_, _path_, _params_, _method_ everything is in different class without good layering.
URI / Request / ...
`Requests` already do the stuff. If 1 day requests stop working (with a ruby X.X) it would be time to vendor it inside our code.

For the moment it seems quite overkill to refactor inside our code a lib without any other dependency.